src/Features/SkySync.cpp

```diff
@@ -0,0 +1,433 @@
+﻿#include "SkySync.h"
+
+NLOHMANN_DEFINE_TYPE_NON_INTRUSIVE_WITH_DEFAULT(
+	SkySync::Settings,
+	Enabled,
+	UseAlternateSunPath,
+	MoonLightSource)
+
+void SkySync::DrawSettings()
+{
+	ImGui::Checkbox("Enabled", &settings.Enabled);
+	ImGui::Checkbox("Use alternate sun path", &settings.UseAlternateSunPath);
+	ImGui::SliderInt("Moon light source", &settings.MoonLightSource, 0, static_cast<uint8_t>(MoonLightSource::Count) - 1, MoonLightSourceNames[settings.MoonLightSource]);
+}
+
+void SkySync::LoadSettings(json& o_json)
+{
+	settings = o_json;
+}
+
+void SkySync::SaveSettings(json& o_json)
+{
+	o_json = settings;
+}
+
+void SkySync::RestoreDefaultSettings()
+{
+	settings = {};
+}
+
+void SkySync::PostPostLoad()
+{
+	moonAndStarsLoaded = GetModuleHandle(L"po3_MoonMod.dll");
+	if (moonAndStarsLoaded)
+		logger::info("[Sky Sync] Moon and Stars detected, compatibility enabled");
+
+	if (GetModuleHandle(L"EVLaS.dll")) {
+		loaded = false;
+		failedLoadedMessage = "Sky Sync has been disabled as EVLaS has been detected, both cannot be used together";
+		logger::warn("{}", failedLoadedMessage);
+		return;
+	}
+
+	stl::detour_thunk<Moon_Update>(REL::RelocationID(25626, 26169));
+	stl::detour_thunk<Sky_Update>(REL::RelocationID(25682, 26229));
+	stl::detour_thunk<Sky_OnNewClimate>(REL::RelocationID(25695, 26242));
+	stl::detour_thunk<VolumetricLightingDescriptor_Get>(REL::RelocationID(100297, 107014));
+
+	gSunPosition = reinterpret_cast<RE::NiPoint3*>(REL::RelocationID(527924, 414871).address());
+	gSunGlareSize = reinterpret_cast<float*>(REL::RelocationID(502611, 370235).address());
+	gMasserSize = reinterpret_cast<uint32_t*>(REL::RelocationID(502558, 370155).address());
+	gSecundaSize = reinterpret_cast<uint32_t*>(REL::RelocationID(502570, 370173).address());
+
+	logger::info("[Sky Sync] Installed hooks");
+}
+
+void SkySync::Sky_Update::thunk(RE::Sky* sky)
+{
+	func(sky);
+	GetSingleton()->Update(sky);
+}
+
+void SkySync::Update(const RE::Sky* sky)
+{
+	if (!settings.Enabled)
+		return;
+
+	const auto sun = sky->sun;
+	const auto climate = sky->currentClimate;
+	const auto player = RE::PlayerCharacter::GetSingleton();
+	if (!sun || !climate || !player)
+		return;
+
+	if (const auto cell = player->GetParentCell(); cell != currentCell) {
+		SetSkyRotation(sky, cell);
+		if (currentCell && (cell->IsInteriorCell() != currentCell->IsInteriorCell() || cell->GetRuntimeData().worldSpace != currentCell->GetRuntimeData().worldSpace))
+			shadowFader.Reset();
+	}
+
+	const float time = sky->currentGameHour;
+	const bool isDayTime = time > timings.sunriseFadeOutMoonEnd && time < timings.sunsetFadeInMoonStart;
```

Comment:
Was the built in function not appropriate?

Reply:
Yeah, this calculation differs slightly. It determines when the sun/moons take lighting priority rather whether it is within daylight hours.